src/Microsoft.VisualStudio.Services.Agent/SecretMasker.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.VisualStudio.Services.Agent
+{
+    [ServiceLocator(Default = typeof(SecretMasker))]
+    public interface ISecretMasker : IAgentService
+    {
+        IEnumerable<ISecretMask> Masks { get; }
+
+        string MaskSecrets(string input);
+
+        void Add(ISecretMask mask);
+    }
+
+    public class SecretMasker : AgentService, ISecretMasker
+    {
+        private const string MaskedString = "********";
+
+        private readonly List<ISecretMask> _secretMasks = new List<ISecretMask>();
```

Comment:
This might need to handle concurrency. The list can get updated when tasks execute (tasks can set additional secrets). We might need a sync lock, I'm not sure whether the concurrent classes handle concurrency while running an enumerator. Actually, the enumerator could be avoided by using a for loop instead of foreach.